spring-pulsar-test/src/test/java/org/springframework/pulsar/test/support/PulsarConsumerTestUtilTests.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright 2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.pulsar.test.support;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
+import static org.assertj.core.api.Assertions.assertThatIllegalArgumentException;
+import static org.springframework.pulsar.test.support.ConsumedMessagesConditions.desiredMessageCount;
+
+import java.time.Duration;
+import java.util.List;
+import java.util.stream.IntStream;
+
+import org.apache.pulsar.client.api.Message;
+import org.apache.pulsar.client.api.PulsarClient;
+import org.apache.pulsar.client.api.PulsarClientException;
+import org.apache.pulsar.client.api.Schema;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import org.springframework.pulsar.core.DefaultPulsarConsumerFactory;
+import org.springframework.pulsar.core.DefaultPulsarProducerFactory;
+import org.springframework.pulsar.core.PulsarTemplate;
+
+/**
+ * Tests for {@link PulsarConsumerTestUtil}.
+ *
+ * @author Jonas Geiregat
+ */
+class PulsarConsumerTestUtilTests implements PulsarTestContainerSupport {
+
+	private PulsarTemplate<Object> pulsarTemplate;
+
+	private DefaultPulsarConsumerFactory<String> pulsarConsumerFactory;
+
+	private static String testTopic(String suffix) {
```

Comment:
Same as [above](https://github.com/spring-projects/spring-pulsar/pull/592/files#r1502001480) 😿 

- I also scoped the topic name a bit more w/ prefix of acronym of the test class name (we do this in other tests as well). 
- I also used the more specific AssertJ "assertThat(IllegalArgument|IllegalState|ExceptionOfType)` APIs